### ISSUE TITLE
Showing an image for JS-Disabled Browsers?

### ISSUE THREAD

**Issue by yakutsa** [scottjehl/picturefill#499]
Is it possible to show an image for JS-disabled browsers?

I know that Picturefill 1.2 supports this feature but I am curious whether Picturefill v2 also supports to show an image when no javascript is enabled.

Is there any solution to this issue?

**Comment by jegtnes** [comment]
Hey @yakutsa,

For critical content you *can* use a `src` attribute as well as a `srcset` attribute, but that will result in a double download in browsers that don't support responsive images natively, but have Picturefill (as at the point Picturefill executes the browser will likely have prefetched the image, if not downloaded it already).

You can also make use of a `<noscript>` tag to display a fallback, but that will result in a double download in browsers which support responsive images natively and can't execute JS for whatever reason. What is probably even worse: They'll both display the responsive image _and_ the fallback at the same time. Which is not ideal. To my knowledge there's no smart way of hiding one of these images in a failproof manner, as feature detection of responsive images, e.g. Modernizr, requires JS. So if you can't rely on that, you'll either end up with two images displaying in browsers with native support, or if you do something like `.no-js noscript > img { display: none; }` you'll fall back to the alt text on browsers without responsive image support as the regular image gets hidden.

So, basically, they're all suboptimal tradeoffs. We [normally recommend merely falling back to the alt text](http://www.filamentgroup.com/lab/to-picturefill.html) for this reason, but if it's a crucial image that you need to always display to browsers with images enabled, you can make use of both `src` and `srcset` and swallow the double download in some browsers, or use Picturefill 1.2. That's up to you, and what's best to do depends on your individual circumstances and your users.

Of course, I'm sure you could do something clever with cookies and server-side code if you're that way inclined. What I'm saying is there's no *nice* way of doing it. :)

**Comment by yakutsa** [comment]
Hi @jegtnes ,

Thank you for detailed explanation.

Picturefill 1.2 doesn't support IE8 nor IE9 so that would be a big tradeoff.  

The main reason I am using Picturefill 2 is that it supports both IE8 and IE9. It is crucial to display the image for JS-disabled browsers so I probably have to go with `<noscript>` tag even though it would result in double download and increase the loading time.

**Comment by quoo** [comment]
Picturefill 1.2 documents fallbacks for IE8: https://github.com/scottjehl/picturefill/tree/1.2#supporting-ie-desktop

It should support IE9 however. Are you having problems with IE9?

**Comment by yakutsa** [comment]
Hi @quoo,

I uploaded my code to a jsfiddle. You can check it: https://jsfiddle.net/y533jfzx/

If the IE version is lower than 10, it just resize itself by `width:100%` . What I want is to get the right image by respective device width. IE9 and IE8 don't support it.

Maybe I am missing something here?

**Comment by QWp6t** [comment]
> The main reason I am using Picturefill 2 is that it supports both IE8 and IE9. It is crucial to display the image for JS-disabled browsers so I probably have to go with `<noscript>` tag even though it would result in double download and increase the loading time.

Terrible idea! Like he said, you will have the image showing up twice in browsers where `srcset` has native support. Use `src` and `srcset` together. You'll still get a double download, but it won't look like shit.

**Comment by aFarkas** [comment]
@yakutsa 
For IE9 support you will also need to include a ``matchMedia`` polyfill (https://cdn.rawgit.com/scottjehl/picturefill/1.2/external/matchmedia.js)

**Comment by yakutsa** [comment]
@QWp6t 

I didn't realize it until I implemented it. Yes, you were right. It was a terrible idea. I am going with `src` and `srcset` as you suggested.

@aFarkas 

I tried that, too. Here's the updated jsfidlee: https://jsfiddle.net/y533jfzx/1/

IE9 still don't act as it should. IE8 doesn't show the image at all.

**Comment by mike-engel** [comment]
@yakutsa, it looks like you have a solution for js disabled browsers. Unless you have any objections, I'm going to close this issue. Thanks!

**Comment by yakutsa** [comment]
@mike-engel ,

Yes, I am closing this issue now. Thank you all to those who contributed to this thread.
